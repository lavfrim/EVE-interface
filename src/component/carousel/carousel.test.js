import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Carousel from './index';

function Component() {
    return (
        <p>
            {'smth'}
        </p>
    );
}

const fractionsComponentArray = [
    <Component key={1}/>,
    <Component key={2} />,
    <Component key={3}/>
];


describe('Carousel should render correctly', () => {
    it('with mock array of components', () => {
        const wrapper = mount(
            <Carousel
                fractionsComponentArray={fractionsComponentArray}
                amountCards={3}
            />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

