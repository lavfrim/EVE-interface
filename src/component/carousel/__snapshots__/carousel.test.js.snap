// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ca should render correctly with error mesage 1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;

exports[`Carousel should render correctly with  1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;

exports[`Carousel should render correctly with error mesage 1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;

exports[`Carousel should render correctly with mock  1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;

exports[`Carousel should render correctly with mock array of compo 1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;

exports[`Carousel should render correctly with mock array of components 1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;

exports[`Main should render correctly with error mesage 1`] = `
<Carousel
  amountCards={3}
  fractionsComponentArray={
    Array [
      <Component />,
      <Component />,
      <Component />,
    ]
  }
>
  <section
    className="carousel"
  >
    <button
      className="carousel__button-backward"
      onClick={[Function]}
    >
      &lt;
    </button>
    <Component
      key="1"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="2"
    >
      <p>
        smth
      </p>
    </Component>
    <Component
      key="3"
    >
      <p>
        smth
      </p>
    </Component>
    <button
      className="carousel__button-forward"
      onClick={[Function]}
    >
      &gt;
    </button>
  </section>
</Carousel>
`;
